aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -317,15 +317,17 @@ public IMyService MyService { get; set; }
+Even in client-side Blazor apps that don't operate over a circuit, services registered with a scoped lifetime are treated as singletons, so they live longer than scoped services in typical ASP.NET Core apps. Client-side disposable transient services also live longer than the components where they're injected because the DI container, which holds references to disposable services, persists for the lifetime of the app, preventing garbage collection on the services. Although long-lived disposable transient services are of greater concern on the server, they should be avoided as client service registrations as well. Use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type is also recommended for client-side scoped services to control service lifetime, and disposable transient services shouldn't be used at all.
 
-An approach that limits a service lifetime is use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type. <xref:Microsoft.AspNetCore.Components.OwningComponentBase> is an abstract type derived from <xref:Microsoft.AspNetCore.Components.ComponentBase> that creates a DI scope corresponding to the *lifetime of the component*. Using this scope, it's possible to use DI services with a scoped lifetime and have them live as long as the component. When the component is destroyed, services from the component's scoped service provider are disposed as well. This can be useful for services that:
+An approach that limits a service lifetime is use of the <xref:Microsoft.AspNetCore.Components.OwningComponentBase> type. <xref:Microsoft.AspNetCore.Components.OwningComponentBase> is an abstract type derived from <xref:Microsoft.AspNetCore.Components.ComponentBase> that creates a DI scope corresponding to the *lifetime of the component*. Using this scope for a scoped service, a component can inject services with a scoped lifetime and have them live as long as the component. When the component is destroyed, services from the component's scoped service provider are disposed as well. This can be useful for services that:
 
-* Should be reused within a component, as the transient lifetime is inappropriate.
+* Should be reused within a component, as the transient lifetime is inappropriate with disposable transient services.
 * Shouldn't be shared across components, as the singleton lifetime is inappropriate.
```

Comment:
I think the original meaning of this part was that the transient lifetime (in general) does not satisfy the "should be reused within a component" requirement. Not sure what you mean by adding the "with disposable transient services" here?

The whole intro to `OwningComponentBase` here is that you can inject a composition root (`ScopedService` or by using `Services`) with lifetime aligned to the component. This is something neither the transient nor the singleton lifetime can offer (when we're talking about regular injection).

Uff, not an easy topic. :-D

Reply:
> Not sure what you mean by adding the "with disposable transient services" here?
>
> neither the transient nor the singleton lifetime can offer (when we're talking about regular injection).

That wasn't my interpretation of Tanay's remarks. BTW, I thought it was Steve that wrote the original design proposal, but it was Tanay, who has since moved to a different team. Anyway, I added "disposable transient services" here because according to Tanay it's ok to use non-disposable transient services. See the first column of his table ...

> Transient Service which do NOT implement IDisposable ... Are garbage collected when the component in which the service was injected is disposed.

https://github.com/dotnet/aspnetcore/issues/26676#issuecomment-1179447445